app/src/endpoints.c

```diff
@@ -78,11 +86,15 @@ int zmk_endpoint_instance_to_str(struct zmk_endpoint_instance endpoint, char *st
     }
 }
 
-#define INSTANCE_INDEX_OFFSET_USB 0
-#define INSTANCE_INDEX_OFFSET_BLE ZMK_ENDPOINT_USB_COUNT
+#define INSTANCE_INDEX_OFFSET_NONE 0
+#define INSTANCE_INDEX_OFFSET_USB ZMK_ENDPOINT_NONE_COUNT
+#define INSTANCE_INDEX_OFFSET_BLE ZMK_ENDPOINT_NONE_COUNT + ZMK_ENDPOINT_USB_COUNT
```

Comment:
```suggestion
#define INSTANCE_INDEX_OFFSET_BLE (ZMK_ENDPOINT_NONE_COUNT + ZMK_ENDPOINT_USB_COUNT)
```
This needs to be parenthesized, or you might get unexpected behaviors like
```c
INSTANCE_INDEX_OFFSET_BLE * 2  // I expect 2 * 2 = 4 but...
ZMK_ENDPOINT_NONE_COUNT + ZMK_ENDPOINT_USB_COUNT * 2  // 1 + 1 * 2 = 3
```

Better yet, each item can be written as the previous offset plus the previous count so the size of the expression doesn't grow with each item:

```c
#define INSTANCE_INDEX_OFFSET_NONE 0
#define INSTANCE_INDEX_OFFSET_USB (INSTANCE_INDEX_OFFSET_NONE + ZMK_ENDPOINT_NONE_COUNT)
#define INSTANCE_INDEX_OFFSET_BLE (INSTANCE_INDEX_OFFSET_USB + ZMK_ENDPOINT_USB_COUNT)
#define INSTANCE_INDEX_OFFSET_FOO (INSTANCE_INDEX_OFFSET_BLE + ZMK_ENDPOINT_BLE_COUNT)
...
```